lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -570,16 +602,15 @@ def safely_acquire_lock_for_table(table, mode: :access_exclusive, &block)
       end
 
       connection.transaction do
-        adjust_timeout_method = connection.postgresql_version >= 9_03_00 ? :adjust_lock_timeout : :adjust_statement_timeout
         begin
-          method(adjust_timeout_method).call(PgHaMigrations::LOCK_TIMEOUT_SECONDS) do
-            connection.execute("LOCK #{target_table.fully_qualified_name} IN #{target_table.mode.to_sql} MODE;")
+          adjust_statement_timeout(PgHaMigrations::LOCK_TIMEOUT_SECONDS) do
```

Comment:
Lock timeout applies to each individual relation in the query, while statement timeout applies to the entire query. So, if we were to use lock timeout here, the upper limit for the query timeout would be `LOCK_TIMEOUT_SECONDS * <number of tables being locked>` which seems incorrect